contracts/0.8.25/vaults/OperatorGrid.sol

```diff
@@ -0,0 +1,595 @@
+    /// @return Node operator count
+    function nodeOperatorCount() external view returns (uint256) {
+        return _getStorage().nodeOperators.length;
+    }
+
+    /// @notice Registers a new tier
+    /// @param _nodeOperator address of the node operator
+    /// @param _tiers array of tiers to register
+    function registerTiers(
+        address _nodeOperator,
+        TierParams[] calldata _tiers
+    ) external onlyRole(REGISTRY_ROLE) {
+        if (_nodeOperator == address(0)) revert ZeroArgument("_nodeOperator");
+
+        ERC7201Storage storage $ = _getStorage();
+        Group storage group_ = $.groups[_nodeOperator];
+        if (group_.operator == address(0)) revert GroupNotExists();
+
+        uint128 tierId = uint128($.tiers.length);
+        uint256 length = _tiers.length;
+        for (uint256 i = 0; i < length; i++) {
+            Tier memory tier_ = Tier({
+                operator: _nodeOperator,
+                shareLimit: uint96(_tiers[i].shareLimit),
+                reserveRatioBP: uint16(_tiers[i].reserveRatioBP),
+                rebalanceThresholdBP: uint16(_tiers[i].rebalanceThresholdBP),
+                treasuryFeeBP: uint16(_tiers[i].treasuryFeeBP),
+                mintedShares: 0
+            });
+            $.tiers.push(tier_);
+            group_.tierIds.push(tierId);
+
+            emit TierAdded(
+                _nodeOperator,
+                tierId,
+                uint96(_tiers[i].shareLimit),
+                uint16(_tiers[i].reserveRatioBP),
+                uint16(_tiers[i].rebalanceThresholdBP),
+                uint16(_tiers[i].treasuryFeeBP)
+            );
+
+            tierId++;
+        }
+    }
+
+    /// @notice Returns a tier by ID
+    /// @param _tierId id of the tier
+    /// @return Tier
+    function tier(uint256 _tierId) external view returns (Tier memory) {
+        return _getStorage().tiers[_tierId];
+    }
+
+    /// @notice Alters a tier
+    /// @dev We do not enforce to update old vaults with the new tier params, only new ones.
+    /// @param _tierId id of the tier
+    /// @param _tierParams new tier params
+    function alterTier(uint256 _tierId, TierParams calldata _tierParams) external onlyRole(REGISTRY_ROLE) {
+        ERC7201Storage storage $ = _getStorage();
+        Tier storage tier_ = $.tiers[_tierId];
+        if (_tierId != DEFAULT_TIER_ID && $.tiers[_tierId].operator == address(0)) revert TierNotExists();
+
+        tier_.shareLimit = uint96(_tierParams.shareLimit);
+        tier_.reserveRatioBP = uint16(_tierParams.reserveRatioBP);
+        tier_.rebalanceThresholdBP = uint16(_tierParams.rebalanceThresholdBP);
+        tier_.treasuryFeeBP = uint16(_tierParams.treasuryFeeBP);
+
+        emit TierUpdated(_tierId, tier_.shareLimit, tier_.reserveRatioBP, tier_.rebalanceThresholdBP, tier_.treasuryFeeBP);
+    }
+
+    /// @notice Registers a new vault
+    /// @param vault address of the vault
+    function registerVault(address vault) external {
```

Comment:
:brain: :brain: :brain: probably can remove this fn completely if we have an agreement that the default tier id is zero